compiler/src/iree/compiler/Codegen/LLVMCPU/test/apply_scale_lowering.mlir

```diff
@@ -0,0 +1,184 @@
+// RUN: iree-opt --split-input-file --pass-pipeline='hal.executable(hal.executable.variant(builtin.module(iree-convert-to-llvm)))' %s | FileCheck %s
+
+#executable_layout = #hal.executable.layout<push_constants = 0, sets = [
+  #hal.descriptor_set.layout<0, bindings = [
+    #hal.descriptor_set.binding<0, storage_buffer>,
+    #hal.descriptor_set.binding<1, storage_buffer>
+  ]>
+]>
+#executable_target_embedded_elf_riscv_64_ = #hal.executable.target<"llvm", "embedded-elf-riscv_64", {
+  cpu_features = "+m,+a,+f,+d,+c",
+  data_layout = "e-m:e-p:64:64-i64:64-i128:128-n64-S128",
+  native_vector_size = 256 : index,
+  target_triple = "riscv64-unknown-unknown-eabi-elf"
+}>
+#map = affine_map<()[s0] -> (s0 ceildiv 2)>
+#translation = #iree_codegen.translation_info<CPUDoubleTilingExpert workload_per_wg = [2]>
+hal.executable private @apply_scale_no_vector_feature {
+  hal.executable.variant public @embedded_elf_riscv_64, target = #executable_target_embedded_elf_riscv_64_ {
+    hal.executable.export public @apply_scale_no_vector_feature ordinal(0) layout(#executable_layout) {translation_info = #translation} {
+    ^bb0(%arg0: !hal.device, %arg1: index, %arg2: index, %arg3: index):
+      %c1 = arith.constant 1 : index
+      %0 = affine.apply #map()[%arg1]
+      hal.return %0, %c1, %c1 : index, index, index
+    }
+    builtin.module {
+      func.func @apply_scale_no_vector_feature() {
+        %cst = arith.constant dense<19689> : vector<2xi32>
+        %cst_0 = arith.constant dense<15> : vector<2xi8>
+        %c0 = arith.constant 0 : index
+        %0 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%c0) alignment(64) : memref<2xi32>
+        %1 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%c0) alignment(64) : memref<2xi32>
+        %2 = vector.load %0[%c0] : memref<2xi32>, vector<2xi32>
+        %3 = "tosa.apply_scale"(%2, %cst, %cst_0) {double_round = false} : (vector<2xi32>, vector<2xi32>, vector<2xi8>) -> vector<2xi32>
+        vector.store %3, %1[%c0] : memref<2xi32>, vector<2xi32>
+        return
+      }
+    }
+  }
+}
+
+// 32-bit lowering should be used by default if no vector features are provided.
```

Comment:
64-bit?

Reply:
oh, I see the logic now. So the intention is not to unroll the scalar-only target, even if the scalar ISA supports int64 (as the target_triple is riscv64 prefixed)?

Reply:
Honestly, I didn't think too much about this case since we would always expect at least some flavor of vector extensions... (otherwise, we shouldn't even vectorized the code!). We can change it to the 64-bit version if you think it's more convenient.

Reply:
If we don't expect any vectorization in this case (no +v, no +zve), shouldn't we check for the pattern of the non-vectorized result? Not sure that is something checkable at the LLVM IR level though.

Reply:
Sorry, my comment was misleading! At this point, we assume that we'll always have some kind of vector support so we always (try to) vectorize the code without actually looking at any target features. The first thing we should do if we wanted to support targets without any vector support is to prevent vectorization to happen, which we don't do at this point. That's why what we do for this case (no +v, no +zve) is not relevant in practice. I just added it to test that corner case in the lowering logic. Hopefully that makes more sense :)

Reply:
Unfortunately that's not how we state in https://google.github.io/iree/building-from-source/riscv/. We probably should not expect everyone who uses IREE + RISCV backend has a v-ext enabled device.

If we currently can't prevent vectorizing for any target that' can't support vectorizing effectively, IMHO, we should create an issue and resolve it, and we probably should add a TODO comment in this test.